### ISSUE TITLE
Call `showWarningMessage` when refmt fails

### ISSUE THREAD

**Issue by jchavarri** [freebroccolo/ocaml-language-server#97]
There are a few cases where it would be good to show errors from refmt:

1. For projects that are not using `bsb`
2. For times where the `bsb` parser version doesn't coincide with the installed `refmt` version

I've run into problems due to 2 and it's very confusing, as the auto-formatting suddenly breaks with no information. I had to check from the command line to see what was going on. This is always an option of course, but we can improve the user experience by parsing the output, with a similar mechanism that we use for `bsb`. 

Except, in this case, it is so much simpler :)

```
File "./src/Presentation.re", line 30, characters 73-74:
Error: 2583: <UNKNOWN SYNTAX ERROR>
```

**Comment by jordwalke** [comment]
This is also pretty helpful from the Atom->ocaml-language-server use case which doesn't have any error feedback when trying to reformat an invalidly written file.
I don't know what the experience in VSCode is like, but in Atom, if you have a syntax error, it's lost in a sea of other potential errors in the Diagnostics panel - so when you hit the key command to reformat and it doesn't tell you that it can't due to a syntax error, it's a little confusing.

**Comment by jchavarri** [comment]
@jordwalke The fix in #99 should help alleviate that pain. When refmt finds an error, it [sends just that one](https://github.com/freebroccolo/ocaml-language-server/pull/99/files#diff-654fa3a98c26f25b183b0d7960e43acdR67) as `diagnostics` of that file, which would "clean" the Diagnostic panel from every other diagnostics from other tools, and allow the user to focus just on that particular one.
